cfme/infrastructure/config_management/__init__.py

```diff
@@ -754,50 +549,52 @@ def step(self, *args, **kwargs):
         row.click()
 
 
-@navigator.register(ConfigManager, 'EditFromDetails')
+@navigator.register(ConfigManagerProvider, 'EditFromDetails')
 class MgrEditFromDetails(CFMENavigateStep):
     VIEW = ConfigManagementEditView
-    prerequisite = NavigateToSibling('Details')
+    prerequisite = NavigateToSibling('AllOfType')
 
     def step(self, *args, **kwargs):
         self.prerequisite_view.toolbar.configuration.item_select('Edit this Provider')
 
 
-@navigator.register(ConfigProfile, 'Details')
-class Details(CFMENavigateStep):
-    VIEW = ConfigManagementProfileView
-    prerequisite = NavigateToAttribute('manager', 'Details')
+@navigator.register(ConfigSystem, 'EditTags')
+class SysEditTags(CFMENavigateStep):
+    VIEW = TagPageView
+    prerequisite = NavigateToAttribute('profile', 'Details')
 
     def step(self, *args, **kwargs):
         self.prerequisite_view.toolbar.view_selector.select('List View')
         row = self.prerequisite_view.entities.paginator.find_row_on_pages(
-            self.prerequisite_view.entities.elements, description=self.obj.name)
-        row.click()
+            self.prerequisite_view.entities.elements, hostname=self.obj.name)
+        row[0].check()
+        self.prerequisite_view.toolbar.policy.item_select('Edit Tags')
 
 
-@navigator.register(ConfigSystem, 'All')
-class SysAll(CFMENavigateStep):
-    VIEW = ConfigSystemAllView
-    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+@navigator.register(ConfigProfile, 'Details')
+class Details(CFMENavigateStep):
+    VIEW = ConfigManagementProfileView
+    prerequisite = NavigateToAttribute('manager', 'Details')
 
     def step(self, *args, **kwargs):
-        if hasattr(self.obj, 'type') and self.obj.type == 'Ansible Tower':
-            self.prerequisite_view.navigation.select('Automation', 'Ansible Tower', 'Explorer')
-            self.view.sidebar.configured_systems.tree.click_path(
-                'All Ansible Tower Configured Systems')
-        else:
-            self.prerequisite_view.navigation.select('Configuration', 'Management')
-            self.view.sidebar.configured_systems.tree.click_path("All Configured Systems")
+        self.prerequisite_view.toolbar.view_selector.select('List View')
+        try:
+            row = self.prerequisite_view.entities.paginator.find_row_on_pages(
+                self.prerequisite_view.entities.elements, name=self.obj.name
+            )
+        except NameError:
+            row = self.prerequisite_view.entities.paginator.find_row_on_pages(
+                self.prerequisite_view.entities.elements, description=self.obj.name
+            )
 
+        row.click()
 
-@navigator.register(ConfigSystem, 'EditTags')
-class SysEditTags(CFMENavigateStep):
-    VIEW = TagPageView
-    prerequisite = NavigateToSibling('All')
+
+@navigator.register(ConfigManagerProviderCollection, "All")
+class ConfigManagerAllPage(CFMENavigateStep):
```

Comment:
```        raise NotImplementedError(
            "There is no page in MIQ that displays all config managers."
            " Use 'AllOfType' on a config manager provider instance."
        )```

The error message provider providers some extra details about what you should do instead. Do you think we should remove that additional info?

Reply:
Mixed feelings - but I do appreciate the extra context provided by the exception here.

I think if it stays, it should raise `navmazing.NavigationDestinationNotFound` instead of `NotImplementedError`.

Reply:
Modified to raise `NavigationDestinationNotFound`

Reply:
Good - I'm willing to accept as some 'tech-debt' to just remove it once callers are familiar with new structure and definition.